doc/source/whatsnew/v1.0.0.rst

```diff
@@ -626,6 +626,52 @@ consistent with the behaviour of :class:`DataFrame` and :class:`Index`.
    DeprecationWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.
    Series([], dtype: float64)
 
+Result dtype inference changes for resample operations
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The rules for the result dtype in :meth:`DataFrame.resample` aggregations have change for extension types (:issue:`31359`).
+Previously, pandas would attempt to convert the result back to the original dtype, falling back to the usual
+inference rules if that was not possible. Now, pandas will only return a result of the original dtype if the
+scalar values in the result are instances of the extension dtype's scalar type.
+
+.. ipython:: python
+
+   df = pd.DataFrame({"A": ['a', 'b']}, dtype='category',
+                     index=pd.date_range('2000', periods=2))
+   df
+
+
+*pandas 0.25.x*
+
+.. code-block:: python
+
+   >>> df.resample("2D").agg(lambda x: 'a').A.dtype
+   CategoricalDtype(categories=['a', 'b'], ordered=False)
+
+*pandas 1.0.0*
+
+.. ipython:: python
+
+   df.resample("2D").agg(lambda x: 'a').A.dtype
+
```

Comment:
What will this return? Not a categorical? 
I am not fully sure that is correct as well. Eg if you do a first-like aggregation `lambda x: x[0]`, you would expect this to work... 

(but of course any heuristic will never be correct in all cases .. for full control in cases like this we will need an additional keyword)

Reply:
Right, it will be object dtype. That's the API breaking change. But, I'm OK with it because

1. It matches the behavior of groupby on 0.25.3. `df.groupby([1, 1]).agg(lambda x: 'a').A.dtype` is object.
2. It's incorrect (or at least surprising?) when the dtype can't hold the values (the next example where you get NaN values).